### PR TITLE
Handle preview click

### PR DESCRIPTION
**Pull request by cadrimiranda**
## Description

Was added a onPreviewClick props to DropZoneAreaBase, and user can click on preview icon to get a callback with file and index of the file object.

## Link related issues

- Feat #231 

## Type of change

New feature: `onPreviewClick` function prop to allow user click on preview icon and the callback this function.

## How Has This Been Tested

I added the onPreviewClick props in all demos and tried to click in all imagens with or without passing one function to prop, and no one's broken or crash. 

- Browser:

Versão 85.0.4183.121 (Versão oficial) 64 bits

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/PreviewList.js [hunk 1]
```diff
@@ -91,6 +91,8 @@ const PreviewList = ({
             className={clsx(previewGridClasses.gridListTile, {
               [classes.iconWrapper]: !isImage,
             })}
+            onClick={handlePreviewClick(i)}
+            onKeyDown={handlePreviewClick(i)}
```
Comment (panz3r) [inline src/components/PreviewList.js:95]: Why did you set a listener on `KeyDown`?

**Review by panz3r — CHANGES_REQUESTED** [review]
Hi @cadrimiranda ,

Thanks for your contribution, LGTM 👍 .

Can you please remove the files under `dist` from the PR?

File: src/components/PreviewList.js (hunk 1 above)
Comment (cadrimiranda) [reply src/components/PreviewList.js:95]: Hello, sorry for my late answer. I added the onKeyDown to allow user navigate with keyboard to increase the accessibility. Is it wrong?

Comment (cadrimiranda) [reply src/components/PreviewList.js:95]: > Hi @cadrimiranda ,
> 
> Thanks for your contribution, LGTM 👍 .
> 
> Can you please remove the files under `dist` from the PR?

Sure, I'll remove.

Comment (cadrimiranda) [reply src/components/PreviewList.js:95]: Done!

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@ build/
 node_modules/
 styleguide/
 .eslintcache
+dist/
```
Comment (panz3r) [inline .gitignore:7]: @cadrimiranda , can you please revert this change?

`dist/` should not be ignored otherwise the compile GitHub action won't work.

**Review by panz3r — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (cadrimiranda) [reply .gitignore]: Sorry, this weekend was a holiday in Brazil and I'm worked almost 24 hours in the last two days. I updated the code and push to git now.

**Review by panz3r — APPROVED** [review]
LGTM 👍
